crates/utils/src/logging.rs

```diff
@@ -53,7 +54,13 @@ pub fn setup_tracing(otel: OpenTelemetry) -> Result<()> {
     let subscriber = Registry::default()
         .with(stdout_layer().with_filter(env_or_default_filter()))
         .with(otel_layer.with_filter(env_or_default_filter()));
-    tracing::subscriber::set_global_default(subscriber).map_err(Into::into)
+    tracing::subscriber::set_global_default(subscriber).map_err(Into::<anyhow::Error>::into)?;
+
+    // Register panic hook now that tracing is initialized.
+    std::panic::set_hook(Box::new(|info| {
+        tracing::error!("{info}");
+    }));
```

Comment:
Should we include something more specific so we can filter for these?

I'm not entirely sure how to do this but maybe we should include a `panic = true` field? Then we can filter by it.